gsrs-module-clinical-trials-spring-boot-autoconfigure/src/main/java/gov/hhs/gsrs/clinicaltrial/us/validators/GeneralTrialOnUpdateValidator.java

```diff
@@ -28,80 +28,75 @@ public boolean supports(ClinicalTrialUS newValue, ClinicalTrialUS oldValue, Vali
+        if (objold == null) {
+            callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(objoldIsNullErrorTemplate)));
+        } else if (objnew == null) {
+            callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(objnewIsNullErrorTemplate)));
         } else {
-            ClinicalTrialUS ct = found.get();
+            String trialNumber = objnew.getTrialNumber();
+            if (trialNumber == null) {
+                callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(trialNumberNullErrorTemplate)));
+            }
+            boolean formatOK = trialNumberPattern.matcher(trialNumber).matches();
+
+            if (!formatOK) {
+                callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(badlyFormattedTrialNumberTemplate, trialNumber)));
+            }
+
             Date newCreationDate = objnew.getCreationDate();
             Date newLastModifiedDate = objnew.getLastModifiedDate();
-            Date oldCreationDate = ct.getCreationDate();
-            Date oldLastModifiedDate = ct.getLastModifiedDate();
+            Date oldCreationDate = objold.getCreationDate();
+            Date oldLastModifiedDate = objold.getLastModifiedDate();
+
+            boolean abort1 = false;
+
             if (newCreationDate == null) {
+                abort1 = true;
                 callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(newCreationDateNullErrorTemplate)));
             }
             if (newLastModifiedDate == null) {
+                abort1 = true;
                 callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(newLastModifiedDateNullErrorTemplate)));
             }
             if (oldCreationDate == null) {
+                abort1 = true;
                 callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(oldCreationDateNullErrorTemplate)));
             }
             if (oldLastModifiedDate == null) {
-                // callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(oldLastModifiedDateNullErrorTemplate)));
+                abort1 = true;
+                callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(oldLastModifiedDateNullErrorTemplate)));
             }
-System.out.println(".... A");
-            LocalDateTime ldt1 = TimeUtil.asLocalDateTime(newLastModifiedDate);
-System.out.println(".... B");
 
-            LocalDateTime ldt2 = TimeUtil.asLocalDateTime(oldLastModifiedDate);
-System.out.println(".... C");
-/*
-            boolean cmp = ldt1.isEqual(ldt2);
-            System.out.println("CMP: "+cmp);
-            System.out.println("new: "+newLastModifiedDate);
-            System.out.println("old: "+oldLastModifiedDate);
-            System.out.println("ld1: "+ldt1.toString());
-            System.out.println("ld2: "+ldt2.toString());
+            if (!abort1) {
```

Comment:
I'd get rid of the abort1 variable, and just check for each of these abort conditions, and add a return to each if statement after adding the error.

The only downside is that you wouldn't log every single issue if there are lots. But I think that's okay.

If you want to keep abort, I'd change the variable  name to just to be "abort". And I'd make the line above this just be "if(abort)return;" and get rid of this if statement here, keeping the rest in the main body. That's largely a flow preference of mine, but I think it's more readable.